### PR TITLE
MOB-3706 Update color list - light and dark mode

### PR DESCRIPTION
**Pull request by AleksanderWedrychowskiKarhoo**
… dark mode.

## JIRA
[MOB-3706](https://karhoo.atlassian.net/browse/MOB-3706)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

**Comment by AleksanderWedrychowskiKarhoo** [comment]
The code is ready, but there is almost no visual impact for the moment, since the default color set is ignored throughout the app. Need to adjust all color calls to use UIKarho.colors.

File: KarhooUISDK.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2075,6 +2080,7 @@
 				407EC73922CF564E00F24CFC /* Attributes.swift */,
 				407EC73822CF564E00F24CFC /* Constraints.swift */,
 				407EC73A22CF564E00F24CFC /* KarhooColors.swift */,
+				518245862790A917006B791A /* KarhooColors+defaultSet.swift */,
```
Comment (dianakarh) [inline KarhooUISDK.xcodeproj/project.pbxproj:2083]: Could you name this KarhooColors+**D**efaultSet? I.e. with an uppercase D

Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK.xcodeproj/project.pbxproj:2083]: Sure

File: KarhooUISDK/CommonUI/Layout/KarhooColors+defaultSet.swift [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    var inactive: UIColor {
+        UIColor.get(lightModeColor: UIColor(hex: "#E0E0E0"), darkModeColor: UIColor(hex: "#464646"))
+    }
+    
+    var warning: UIColor {
+        UIColor.get(lightModeColor: UIColor(hex: "#F6A705"), darkModeColor: UIColor(hex: "#FAF06E"))
+    }
+
+    /// White color fit for both loght and dark mode.
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/Layout/KarhooColors+defaultSet.swift:52]: You have a typo for light. Same for the description for the black color.

File: KarhooUISDK/CommonUI/Layout/KarhooColors.swift [hunk 3]
```diff
@@ -9,120 +9,105 @@
...
+
+    var background6: UIColor { get }
+
+    var background7: UIColor { get }
+
+    // MARK: - Text Colors
+
+    var text: UIColor { get }
+
+    var textLabel: UIColor { get }
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/Layout/KarhooColors.swift:60]: What's the difference between text and textLabel?

File: KarhooUISDK/Extensions/UIColor+extension.swift [hunk 4]
```diff
@@ -16,29 +16,56 @@ extension UIColor {
...
+                assertionFailure()
             }
         }
+    }
 
-        return nil
+    static func get(lightModeColor: UIColor, darkModeColor: UIColor) -> UIColor {
+        if #available(iOS 13, *) {
+            return UIColor { (traitCollection) -> UIColor in
+                return traitCollection.userInterfaceStyle == .light ? lightModeColor : darkModeColor
```
Comment (dianakarh) [inline KarhooUISDK/Extensions/UIColor+extension.swift:65]: There's also .unspecified. Could you check with Olivier what's the default in that case? We may need to change the if to `if style == .dark ? darkModeColor : lightModeColor` (depending on his answer)

File: KarhooUISDKTests/TestCases/Extension/UIColorHexSpec.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+        XCTAssert(UIColor(hex: hex) == expectedColor)
+    }
+
+    func testRandomHexInitialization() {
+        let randomHex = "#12dasa"
+        
+        _ = UIColor(hex: randomHex)
+
+        // Thest succeded since assertion has been checked. In case of failure hex initialization would have stoped the scope execution.
```
Comment (dianakarh) [inline KarhooUISDKTests/TestCases/Extension/UIColorHexSpec.swift:27]: Typo in the comment

File: KarhooUISDK/CommonUI/Layout/KarhooColors+DefaultSet.swift [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+    var inactive: UIColor {
+        UIColor.get(lightModeColor: UIColor(hex: "#E0E0E0"), darkModeColor: UIColor(hex: "#464646"))
+    }
+    
+    var warning: UIColor {
+        UIColor.get(lightModeColor: UIColor(hex: "#F6A705"), darkModeColor: UIColor(hex: "#FAF06E"))
+    }
+
+    /// White color fit for both loght and dark mode.
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/Layout/KarhooColors+DefaultSet.swift:52]: Typo here as well. Both on black and white descriptions

**Review by dianakarh — APPROVED** [review]

**Review by mohadian — APPROVED** [review]

**Pull request merged**